### PR TITLE
docs: api/grn_table_cursor: Move the grn_table_cursor_open reference to the header file

### PR DESCRIPTION
**Pull request by abetomo**
GitHub: GH-1817

Prepare to switch to documents automatically generated by Doxygen.

### CODE REVIEW

File: include/groonga/table.h [hunk 1]
```diff
@@ -114,6 +114,81 @@ grn_table_truncate(grn_ctx *ctx, grn_obj *table);
...
+ * \param offset Extracts records from the range of records that meet the
+ *               condition, starting with the `offset`-th (offset is
+ *               zero-based). When GRN_CURSOR_PREFIX is specified, negative
+ *               numbers cannot be specified
+ * \param limit Only `limit` records in the range that meet the condition are
+ *              extracted. -1 means all.
+ *              When GRN_CURSOR_PREFIX is specified, negative numbers cannot
+ *              be specified
+ * \param flags
+ *     * GRN_CURSOR_ASCENDING: Retrieve records in ascending order
```
Comment (otegami) [inline include/groonga/table.h:138]: 📝 List style
- https://www.doxygen.nl/manual/lists.html

File: include/groonga/table.h [hunk 2]
```diff
@@ -114,6 +114,81 @@ grn_table_truncate(grn_ctx *ctx, grn_obj *table);
...
+ *       * If `max` is `NULL`, or if GRN_CURSOR_PREFIX is specified, it will
+ *         be ignored
+ *     * GRN_CURSOR_BY_ID: Retrieves records in ID order
+ *       * If GRN_CURSOR_PREFIX is specified, it will be ignored
+ *     * GRN_CURSOR_BY_KEY: Retrieves records in key order
+ *       * It can be used with table that specify GRN_OBJ_TABLE_PAT_KEY
+ *       * For table with GRN_OBJ_TABLE_HASH_KEY or GRN_OBJ_TABLE_NO_KEY, it
+ *         will be ignored
+ *     * GRN_CURSOR_PREFIX: A cursor is created to retrieve the following
+ *       records for the table GRN_OBJ_TABLE_PAT_KEY
```
Comment (otegami) [inline include/groonga/table.h:160]:
```suggestion
 *       records for the table with GRN_OBJ_TABLE_PAT_KEY
```

**Review by otegami — COMMENTED** [review]
It looks nice to me although I did a few comments.

**Review by otegami — APPROVED** [review]
LGTM 👍🏾

File: include/groonga/table.h [hunk 3]
```diff
@@ -114,6 +114,81 @@ grn_table_truncate(grn_ctx *ctx, grn_obj *table);
...
+ *               zero-based). When GRN_CURSOR_PREFIX is specified, negative
+ *               numbers cannot be specified
+ * \param limit Only `limit` records in the range that meet the condition are
+ *              extracted. -1 means all.
+ *              When GRN_CURSOR_PREFIX is specified, negative numbers cannot
+ *              be specified
+ * \param flags
+ *     * GRN_CURSOR_ASCENDING: Retrieve records in ascending order
+ *       * If GRN_CURSOR_PREFIX is specified and a record with a near key is
+ *       retrieved, or a common prefix search is used, it will be ignored
```
Comment (kou) [inline include/groonga/table.h:140]:
```suggestion
 *         retrieved, or a common prefix search is used, it will be ignored
```

**Pull request merged**